### ISSUE TITLE
Cloud functions creation failed due to npm ci command

### ISSUE THREAD

**Issue by onkarcraft** [GoogleCloudPlatform/datashare-toolkit#634]
**Describe the bug**
We are trying to install datashare. During a terraform deploy stage Cloud Function creation is failing with below message:

"message": "Build failed: npm ERR! The `npm ci` command can only install with an existing package-lock.json or\nnpm ERR! npm-shrinkwrap.json with lockfileVersion \u003e= 1. Run an install with npm@5 or\nnpm ERR! later to generate a package-lock.json file, then try again.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /www-data-home/.npm/_logs/2022-06-08T10_34_15_250Z-debug-0.log; Error ID: beaf8772"

We verified that datashare-batch-cloud-function-src.zip file is present on the GCP bucket and package-lock.json is also present inside the zip file.

**Expected behavior**
Terraform deploy stage should succeed

**Additional context**
We are running this terraform on CircleCI ubuntu image.

**Comment by mservidio** [comment]
Are you able to try and run it from Cloud Shell?

**Comment by malan-craft** [comment]
Hey, thanks for the quick turnaround. In case you want how this error comes up in CircleCI:

```
Error: Error waiting for Creating CloudFunctions Function: Error code 3, message: Build failed: npm ERR! The npm ci command can only install with an existing package-lock.json or
npm ERR! npm-shrinkwrap.json with lockfileVersion >= 1. Run an install with npm@5 or
npm ERR! later to generate a package-lock.json file, then try again.

npm ERR! A complete log of this run can be found in:
npm ERR!     /www-data-home/.npm/_logs/2022-06-08T10_34_15_250Z-debug-0.log; Error ID: beaf8772

  on ../../modules/ingestion-function/cloud-functions.tf line 60, in resource "google_cloudfunctions_function" "datashare_cloud_function":
  60: resource "google_cloudfunctions_function" "datashare_cloud_function" {
```

We were able to do it via local terminal(place zip and run npm ci) and we can try via cloud shell. 

But `npm ci` command is not part of repo and I believe this error is originating from the [`datashare_cloud_function`](https://github.com/GoogleCloudPlatform/datashare-toolkit/blob/master/terraform/modules/ingestion-function/cloud-functions.tf#L56) resource block. If we run this command manually - we will still face issue when terraform tries to create this cloud function resource. How do we go about this? 

We are not sure if running this via cloud shell will help us in the long run as we would like this to be automated as much as possible.

**Comment by mservidio** [comment]
It may have partially deployed with an error. If you look in the GCP console at the Cloud Functions do you see the function created in error? If so, check the log for errors.

**Comment by malan-craft** [comment]
I can see the cloud function saying 'Function failed to deploy', the logs show the same error thrown out by terraform 


`Build failed: npm ERR! The `npm ci` command can only install with an existing package-lock.json or
npm ERR! npm-shrinkwrap.json with lockfileVersion >= 1. Run an install with npm@5 or
npm ERR! later to generate a package-lock.json file, then try again.

npm ERR! A complete log of this run can be found in:
npm ERR!     /www-data-home/.npm/_logs/2022-06-08T10_34_15_250Z-debug-0.log; Error ID: beaf8772`

How do we know if there is a partial deployment happened?

**Comment by malan-craft** [comment]
Yep, I checked here previously and I can see it says 'Function failed to deploy'. The error from the logs is added in above comment - Now that we know there is a partial deployment happening due to NPM error, how do we handle the cloud function creation/deployment via terraform?

**Comment by mservidio** [comment]
For now, you can comment out the resource for the cloud function creation in cloud-functions.tf. I'm not sure what your full requirements are and if you'll need it. Let's see if there are any subsequent issues beyond the cloud function creation within the environment. Is this for marketplace integration?

resource "google_cloudfunctions_function" "datashare_cloud_function"

**Comment by onkarcraft** [comment]
We were able to  complete the terraform steps after commenting out "datashare_cloud_function" but we don't see any datashare instance on GCP account.  We completed the steps mentioned in [INSTALLING.md](  https://github.com/GoogleCloudPlatform/datashare-toolkit/blob/master/INSTALLING.md) this file except for the domain verification. What's the sequence of steps we need to do? There are numerous md files and each has few steps.
